src/JMS/Serializer/XmlDeserializationVisitor.php

```diff
@@ -346,4 +346,34 @@ public function getDoctypeWhitelist()
     {
         return $this->doctypeWhitelist;
     }
+
+    /**
+     * Retrieves internalSubset even in bugfixed php versions
+     *
+     * @param \DOMDocumentType $child
+     * @param string $data
+     * @return string
+     */
+    private function getDomDocumentTypeEntitySubset(\DOMDocumentType $child, $data)
+    {
+        if((PHP_VERSION_ID >= 50513) || (PHP_VERSION_ID >= 50429 && PHP_VERSION_ID < 50500)){
```

Comment:
Instead of checking for all the versions here, can we check for whether ``$child->internalSubset`` contains the data that we expect? That seems more robust to me.